test_runner/src/main/kotlin/ftl/args/yml/YamlDeviceFix.kt

```diff
@@ -0,0 +1,37 @@
+package ftl.args.yml
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+import ftl.args.ArgsHelper.yamlMapper
+import ftl.run.exception.FlankGeneralError
+import ftl.util.loadFile
+import java.nio.file.Path
+
+fun fixDevices(yamlPath: Path) =
+    if (yamlPath.toFile().exists().not()) throw FlankGeneralError("Flank yml doesn't exist at path $yamlPath")
+    else yamlPath.loadYml().fixDevices().saveFixedYml(yamlPath)
+
+private fun Path.loadYml() = yamlMapper.readTree(loadFile(this))
+
+private fun JsonNode.fixDevices() = also {
+    getDevicesNode()?.getNotValidDevices()?.forEach { device ->
+        device.fixDeviceVersion()
+    }
+}
+
+internal fun JsonNode.getDevicesNode() = get(GCLOUD_NODE).get(DEVICES_NODE)
+
+internal fun JsonNode.getNotValidDevices() = filter { it.deviceVersionValid().not() }
```

Comment:
How about `filterNot {...}` instead. We could get rid of `.not()`

Reply:
👍